idm/idm.cpp

```diff
@@ -0,0 +1,221 @@
+        }
+        QTimer::singleShot(200, this, [this] {
+            m_modbusMaster->readHoldingRegister(Idm::ModbusUnitID, Idm::TargetHotWaterTemperature, 1);
+        });
+        break;
+    case Idm::TargetHotWaterTemperature:
+        /* qCDebug(dcIdm()) << "received target hot water temperature"; */
+        if (value.length() == 1) {
+            /* The hot water target temperature is stored as UCHAR (manual p. 13) */
+            m_info->targetWaterTemperature = (double)value[RegisterList::TargetHotWaterTemperature - modbusRegister];
+        }
+        QTimer::singleShot(200, this, [this] {
+            m_modbusMaster->readHoldingRegister(Idm::ModbusUnitID, Idm::HeatPumpOperatingMode, 1);
+        });
+        break;
+    case Idm::HeatPumpOperatingMode:
+        /* qCDebug(dcIdm()) << "received heat pump operating mode"; */
+        if (value.length() == 1) {
+            m_info->mode = heatPumpOperationModeToString((Idm::IdmHeatPumpMode)value[RegisterList::HeatPumpOperatingMode-modbusRegister]);
+        }
+        QTimer::singleShot(200, this, [this] {
+            m_modbusMaster->readHoldingRegister(Idm::ModbusUnitID, Idm::RoomTemperatureHKA, 2);
+        });
+        break;
+    case Idm::RoomTemperatureHKA:
+        /* qCDebug(dcIdm()) << "received room temperature hka"; */
+        if (value.length() == 2) {
+            m_info->roomTemperature = ModbusHelpers::convertRegisterToFloat(&value[RegisterList::RoomTemperatureHKA - modbusRegister]);
+        }
+        QTimer::singleShot(200, this, [this] {
+            m_modbusMaster->readHoldingRegister(Idm::ModbusUnitID, Idm::RoomTemperatureTargetHeatingEcoHKA, 2);
+        });
+        break;
+    case Idm::RoomTemperatureTargetHeatingEcoHKA:
+        /* qCDebug(dcIdm()) << "received room temprature hka eco"; */
+        if (value.length() == 2) {
+            m_info->targetRoomTemperature = ModbusHelpers::convertRegisterToFloat(&value[RegisterList::RoomTemperatureTargetHeatingEcoHKA - modbusRegister]);
+        }
+        QTimer::singleShot(200, this, [this] {
+            m_modbusMaster->readHoldingRegister(Idm::ModbusUnitID, Idm::CurrentPowerConsumptionHeatPump, 2);
+        });
+        break;
+    case Idm::CurrentPowerConsumptionHeatPump:
+        /* qCDebug(dcIdm()) << "received power consumption heat pump"; */
+        if (value.length() == 2) {
+            m_info->powerConsumptionHeatPump = ModbusHelpers::convertRegisterToFloat(&value[RegisterList::CurrentPowerConsumptionHeatPump - modbusRegister]);
+        }
+
+        /* Everything read without an error
+         * -> set connected to true */
+        m_info->connected = true;
+        emit statusUpdated(m_info);
+        break;
+    }
+}
+
+void Idm::onModbusError()
+{
+    qCDebug(dcIdm()) << "iDM: Received modbus error";
+
+    if (m_info != nullptr) {
+        m_info->connected = false;
+        emit statusUpdated(m_info);
+    }
+}
+
+void Idm::onRequestStatus()
+{
+    m_info = new IdmInfo;
```

Comment:
You are leaking memory here, you reserve new memory on the heap but never clean it up. 
Also: do you need to allocate the struct with new here? A private member of the struct should do the job I guess? If you stay with the new, you should also delete it in this class. 

You could also provide a idmInfo() method and use that in the onStatuReceived

Reply:
changed to non pointer